x/twap/api_test.go

```diff
@@ -26,63 +41,87 @@ var (
 	tPlus10sp5Record = newTwapRecordWithDefaults(
 		baseTime.Add(10*time.Second), sdk.NewDec(5), accum0, accum1)
 
+	tPlus10sp5TapRecordAB, tPlus10sp5TapRecordAC, tPlus10sp5TapRecordBC = newThreeAssetPoolTwapRecordWithDefaults(
+		baseTime.Add(10*time.Second), sdk.NewDec(5), accum0, accum1)
+
 	// accumulators updated from tPlus10sp5Record with
 	// t = baseTime + 20
 	// sp0 = 2, sp1 = 0.5
 	tPlus20sp2Record = newTwapRecordWithDefaults(
 		baseTime.Add(20*time.Second), sdk.NewDec(2), OneSec.MulInt64(10*10+5*10), OneSec.MulInt64(3))
+
+	tPlus20sp2TapRecordAB, tPlus20sp2TapRecordAC, tPlus20sp2TapRecordBC = newThreeAssetPoolTwapRecordWithDefaults(
+		baseTime.Add(20*time.Second), sdk.NewDec(2), OneSec.MulInt64(10*10+5*10), OneSec.MulInt64(3))
 )
 
 func (s *TestSuite) TestGetBeginBlockAccumulatorRecord() {
 	poolId, denomA, denomB := s.setupDefaultPool()
 	initStartRecord := newRecord(poolId, s.Ctx.BlockTime(), sdk.OneDec(), sdk.ZeroDec(), sdk.ZeroDec())
-	initStartRecord.PoolId, initStartRecord.Height = poolId, s.Ctx.BlockHeight()
-	initStartRecord.Asset0Denom, initStartRecord.Asset1Denom = denomB, denomA
-
+	initStartRecord[0].Height = s.Ctx.BlockHeight()
 	zeroAccumTenPoint1Record := recordWithUpdatedSpotPrice(initStartRecord, sdk.NewDec(10), sdk.NewDecWithPrec(1, 1))
 
+	tapPoolId, tapDenomA, tapDenomB, tapDenomC := s.setupDefaultThreeAssetPool()
+	tapInitStartRecord := newTapRecord(tapPoolId, s.Ctx.BlockTime(), sdk.OneDec(), sdk.ZeroDec(), sdk.ZeroDec())
+	for i := range tapInitStartRecord {
+		tapInitStartRecord[i].Height = s.Ctx.BlockHeight()
+	}
+	tapZeroAccumTenPoint1Record := recordWithUpdatedSpotPrice(tapInitStartRecord, sdk.NewDec(10), sdk.NewDecWithPrec(1, 1))
+
+	denomAB := [][]string{{denomA}, {denomB}}
```

Comment:
I think two dimensional array can be replaced with a single dimension array here since all the test cases only have one element per array

Reply:
I think that would be the case if we didn't have to call this as well 

https://github.com/osmosis-labs/osmosis/blob/2114b00d399c78176463707380103b9a66afb11b/x/twap/api_test.go#L73